### PR TITLE
Fix display of category from other shop

### LINKED ISSUE #31797: Disallow display of category from other shop
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

With a multishop installation, you can display the list of products of a category from another store, even if this category is not associated with the current store.  
Just need to know the category ID
ex: https://shop.com/1234-id-cat-other-shop

### Expected behavior

display a 404 page if the category is not associated with the current shop

### Steps to reproduce

1) you have a multisite installation with two shop (ore more)
2) in general context create a category tree 
- root
-- shop 1
--- cat-1
-- shop 2
--- cat-2
3) got to shop 1 assign cat-1 to this shop  
4) got to shop 2 assign cat-2 to this shop  

Get the id of cat-1
Got to front of shop 2, on listing of cat-2, 
change the id of cat-2 with id cat-1   
You will see category of shop-1 on shop-2

### PrestaShop version(s) where the bug happened

1.7.8.8

### PHP version(s) where the bug happened

7.4

### If your bug is related to a module, specify its name and its version

_No response_

### PR DESCRIPTION
**Pull request by mkdgs**
Display a 404 error page when another shop's category is called see #31797

| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           |  8.0.x
| Description?      | Please be specific when describing the PR. <br> Every detail helps: versions, browser/server configuration, specific module/theme, etc. Feel free to add more information below this table.
| Type?             | bug fix / improvement / new feature / refacto
| Category?         | FO
| BC breaks?        |  no
| Deprecations?     |  no
| How to test?      | see issue related
| Fixed ticket?     | Fixes 
| Related PRs       | 
| Sponsor company   |

### CODE REVIEW

File: controllers/front/listing/CategoryController.php [hunk 1]
```diff
@@ -80,7 +80,7 @@ public function init()
 
         parent::init();
 
-        if (!Validate::isLoadedObject($this->category) || !$this->category->active) {
+        if (!Validate::isLoadedObject($this->category) || !$this->category->active || $this->category->existsInShop($this->context->shop->id)) {
```
Comment (kpodemski) [inline controllers/front/listing/CategoryController.php]:
```suggestion
        if (!Validate::isLoadedObject($this->category) || !$this->category->active || !$this->category->existsInShop($this->context->shop->id)) {
```

I think you wanted to check whether it doesn't exist. Right?

Comment (kpodemski) [reply controllers/front/listing/CategoryController.php]: also, I think you could send this fix against 8.0.x branch

I'm pretty sure @Hlavtox fixed it some time ago, not sure what happened that it's still a problem 🤔 any hints Daniel?

Comment (mkdgs) [reply controllers/front/listing/CategoryController.php]: Note, breadcrumbs on concerned 404 page also need to be fixed.
The full path of the other shop category is leaked

**Comment by mkdgs** [comment]
Note, breadcrumbs on concerned 404 page also need to be fixed.
The full path of the other shop category is leaked

File: controllers/front/listing/CategoryController.php (hunk 1 above)
Comment (Hlavtox) [reply controllers/front/listing/CategoryController.php]: @kpodemski I think it was customer groups that we fixed or improved, there was also a wrong var used. But nothing with multistore. :-)

But from looking at it, it's a legit fix.

**Review by Hlavtox — DISMISSED** [review]
LGTM

I would handle the breadcrumbs in different PR possibly, I think the same scenario could happen with products maybe?

**Comment by mkdgs** [comment]
I agree, that needs to be handled separately for the breadcrumbs.
If some leaf of category tree are shared across shops, breadcrumbs will probably have the same kind of problem. 
I think, anything relying on Category::getAllParents() will be have a problem in multishop.

**Comment by kpodemski** [comment]
@mkdgs I think this is the bug that should be fixed as soon as possible. Could you change the branch to 8.0.x? this way, we could have your fix inside 8.0.3

**Comment by mkdgs** [comment]
@kpodemski, if this one is merged i think is better to make new PR for 8.0.x
When i change the branch i have tons of commits waiting for merge

**Comment by SharakPL** [comment]
Just update your `origin/8.0.x` branch and rebase your PR on top of it. This will avoid all unrelated commits.

**Comment by MatShir** [comment]
@mkdgs you have change back the pr to `develop` :/

**Comment by mkdgs** [comment]
@MatShir, @SharakPL moved on #31876
Pr was on develop, I have cherry-picked the commit on 8.0.x.
Definitely i don't know how to do this without creating a new branch :/

**Pull request closed without merging**